actors/runtime/src/main/java/cloud/orbit/actors/runtime/Hosting.java

```diff
@@ -400,17 +397,7 @@ private void setCachedAddress(final RemoteReference<?> actorReference, final Nod
 
     private ConcurrentMap<RemoteKey, NodeAddress> getDistributedDirectory()
     {
-        if (distributedDirectory == null)
-        {
-            synchronized (this)
-            {
-                if (distributedDirectory == null)
-                {
-                    distributedDirectory = clusterPeer.getCache("distributedDirectory");
-                }
-            }
-        }
-        return distributedDirectory;
+        return clusterPeer.getCache("distributedDirectory");
```

Comment:
This is a really hot code path, do you have any concerned about the overhead for Infinispan to grab it?
Just wondering if we really don't want to cache this or if the right solution is to clear it during shutdown.

Thoughts?